CMakeLists.txt

```diff
@@ -44,11 +44,13 @@ set(TEST_LIBS ${Boost_UNIT_TEST_FRAMEWORK_LIBRARY} ${Boost_LIBRARIES})
 
 # Workaround
 if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-  if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.5.0)
-  else()
+  if (CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL 3.5.0)
```

Comment:
Ross just [tested it](https://maidsafe.atlassian.net/browse/MAID-828). The 3.6 clang reports different issue, fixed [here](https://github.com/cruxorg/MaidSafe-CRUX/commit/84d399d2ff43c8d4c96042efa66ceb6a7a271b71) I think. I'll do a cherry pick into this PR and we'll see.